pipeline/astro_cal.py

```diff
@@ -28,11 +149,472 @@ class AstroCalibrator:
     def __init__(self, **kwargs):
         self.pars = ParsAstroCalibrator(**kwargs)
 
-    def run(self, *args, **kwargs):
+    # ----------------------------------------------------------------------
+
+    def download_GaiaDR3( self, minra, maxra, mindec, maxdec, padding=0.1, minmag=18., maxmag=22. ):
+        """Download objects from GaiaDR3 as served by Noirlab Astro Data Lab
+
+        Will Get a square on the sky given the limits, with the limits
+        fractionally expanded at each edge by padding.
+
+        minra: float
+           Minimum ra of the image we're trying to match.
+        maxra: float
+           Maximum ra of the image we're trying to match.
+        mindec: float
+           Minimum dec of the image we're trying to match.
+        maxdec: float
+           Maximum dec of the image we're tryhing to match.
+        padding: float
+           A fraction of the width/height by which we expand the area searched on each side.
+        minmag: float
+           Only get starts dimmer than this g magnitude.  (Useful for images that are
+           deep enough that brighter stars will have saturated.)  Make it None to
+           turn off the limit.
+        maxmag: float
+           Only get stars brigther than this g magnitude.  (Useful for images that aren't
+           as deep as Gaia, or for galacitc fields that have huge numbers of stars.)  Make
+           it None to turn off the limit.
+
+        Returns
+        -------
+        CatalogExcerpt
+          A new CatalogExceprt object
+        str
+          The absolute path where the catalog was saved
+        str
+          The absolute path to where the catalog should be saved if it is to be saved to the file store
+
+        Filename is the path where the file was stored; pass this to save() to
+        properly save things to the database.
+
         """
-        Extract sources and use their positions to calculate the astrometric solution.
-        Arguments are parsed by the DataStore.parse_args() method.
 
+        # Sanity check
+        if maxra < minra:
+            tmp = maxra
+            maxra = minra
+            minra = tmp
```

Comment:
...I hadn't thoguht about it.  There are other places we have to deal with this too, including notably in FourCorners.

I just created Issue #111.  This is something we should think through pretty carefully.  Since it's bigger than just astrometry, it's probably something we should do in its own PR.